lib/metadata/wrapper.js

```diff
@@ -67,54 +67,55 @@ const metadata = {
         });
     },
 
-    putObjectMD: (bucketName, objName, objVal, log, cb, params) => {
+    putObjectMD: (bucketName, objName, objVal, params, log, cb) => {
         log.debug('putting object in metdata');
-        client.putObject(bucketName, objName, objVal, log, err => {
+        client.putObject(bucketName, objName, objVal, params, log, err => {
```

Comment:
All versioning modifications (`PUT`/`DELETE`) return some information (such as `versionId`). Please put additional `data` to the callback of `client.putObject`. Or better off, change the whole function to:
```js
putObjectMD: (bucketName, objName, objVal, params, log, cb) => {
    log.debug('putting object in metdata');
    client.putObject(bucketName, objName, objVal, params, log, cb);
}
```